op-service/sources/l1_beacon_client.go

```diff
@@ -209,7 +209,7 @@ func (cl *L1BeaconClient) fetchSidecars(ctx context.Context, slot uint64, hashes
 	for i := 0; i < cl.pool.Len(); i++ {
 		f := cl.pool.Get()
 		resp, err := f.BeaconBlobSideCars(ctx, cl.cfg.FetchAllSidecars, slot, hashes)
```

Comment:
I think it would be better if this check happened inside of `BeaconBlobSideCars`, seems like a leaky abstraction to do it outside

Reply:
Good point, fixed!